packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -571,6 +573,7 @@ class HeroController extends NavigatorObserver {
   }
 
   void _updateQuest(Duration timeStamp) {
+    assert(!_questsEnabled);
     if (navigator == null) {
       // The navigator was removed before this end-of-frame callback was called.
       _clearPendingHeroQuest();
```

Comment:
If the Navigator is disposed before the Hero quest is set up, and then the existing HeroController is reused with a new Navigator, then yes.  I'll reset the _questsEnabled flag in _clearPendingRequests, it makes a bit more sense there.